src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newModifiedArr = [];
```

Comment:
[NAMING] - use proper names for object copy. The variable `newModifiedArr` is not descriptive of its contents. A better name would be `stateHistory` or `statesAfterActions` to reflect that it contains the history of states after each action.